### PR TITLE
Support both Prost and rust-protobuf

### PR DESCRIPTION
**Pull request by nrc**
This is an alternative to https://github.com/pingcap/raft-rs/pull/260. It does less, hopefully the minimum required to get the 0.4.x branch to support both prost and rust-protobuf. It effectively merges the 0.4.x and prost branches. I don't merge to master since TiKV (and thus kvproto) are using the 0.4.x branch, so this is what is needed to permit development on master of kvproto.

### CODE REVIEW

**Comment by nrc** [comment]
This also moves the 0.4.x branch to Rust 2018, since the Prost (and master) branches are there already.

File: README.md [hunk 1]
```diff
@@ -29,6 +29,10 @@ A complete Raft model contains 4 essential parts:
 
 > Note: This Raft implementation in Rust includes the core Consensus Module only, not the other parts. The core Consensus Module in the Raft crate is customizable, flexible, and resilient. You can directly use the Raft crate, but you will need to build your own Log, State Machine and Transport components.
 
+## Using the raft crate
+
+You can use raft with either rust-protobuf or Prost to encode/decode gRPC messages. We use rust-protobuf by default. To use Prost, build (or depend on) Raft using the `prost-codec` feature and without default features.
```
Comment (Hoverbear) [inline README.md]:
```suggestion
You can use raft with either [rust-protobuf](https://github.com/pingcap/rust-protobuf) or [Prost](https://github.com/danburkert/prost) to encode/decode gRPC messages. We use rust-protobuf by default. To use Prost, build (or depend on) Raft using the `prost-codec` feature and without default features.
```

**Comment by nrc** [comment]
Hmm, I guess since we didn't build the protos before we didn't need protoc in CI...

**Comment by Hoverbear** [comment]
@nrc Please close this if it is stale

**Comment by BusyJay** [comment]
Close due to #264.

**Pull request closed without merging**